### PR TITLE
Hide intercom chat in production

### PR DESCRIPTION
**Pull request by Irma12**
Preview https://deploy-preview-1311--roadie.netlify.app/

Ticket: https://app.shortcut.com/larder/story/21209/intercom-launcher-on-marketing-site-opens-automatically

### CODE REVIEW

File: gatsby-ssr.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const onRenderBody = ({ setPostBodyComponents }) => {
...
     };
 
-    (function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',w.intercomSettings);}else{var d=document;var i=function(){i.c(arguments);};i.q=[];i.c=function(args){i.q.push(args);};w.Intercom=i;var l=function(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/${INTERCOM_APP_ID}';var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);};if(document.readyState==='complete'){l();}else if(w.attachEvent){w.attachEvent('onload',l);}else{w.addEventListener('load',l,false);}}})();
+    (function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',intercomSettings);}else{var d=document;var i=function(){i.c(arguments)};i.q=[];i.c=function(args){i.q.push(args)};w.Intercom=i;function l(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/${INTERCOM_APP_ID}';var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);}function ld(){setTimeout(l, 0);}if(w.attachEvent){w.attachEvent('onload',ld);}else{w.addEventListener('load',ld,false);}}})();
```
Comment (dtuite) [inline gatsby-ssr.js:14]: What do these changes do?

File: gatsby-browser.js [hunk 2]
```diff
@@ -24,3 +24,27 @@ import './src/stylesheets/swagger-ui.css';
...
+const isHidden = () => process.env.NODE_ENV === `production` && typeof Intercom === `function`;
+
+const onInitialClientRender = () => {
+  if (!isHidden()) {
+    return;
+  }
+
+  window.Intercom('boot', {
+    app_id: INTERCOM_APP_ID,
+    hide_default_launcher: isHidden,
```
Comment (dtuite) [inline gatsby-browser.js:38]: Can you make sure Leena gets to see the preview please? I'm not sure if this is what she intended or not.

**Review by dtuite — APPROVED** [review]

File: gatsby-ssr.js (hunk 1 above)
Comment (Irma12) [reply gatsby-ssr.js:14]: Ah this is leftover from something I tried before, I will revert this

File: gatsby-browser.js (hunk 2 above)
Comment (Irma12) [reply gatsby-browser.js:38]: Checked, she confirmed it works as expected

**Pull request merged**